src/build.ts

```diff
@@ -0,0 +1,23 @@
+
+// expected injected enviornment variables
+declare const GTM: any;
+declare const SENTRY: any;
+declare const PROD_MODE: any;
+
+interface BuildConfig {
+  enableGoogleTagManager: boolean;
```

Comment:
Nit picking here, but I personally think starting with `enableFoo` suggests it's an action when in fact it's intention is to be a flag. I think `googleTagManagerEnabled` would be more clear. Or you could create a nested key likes so:

```ts
{
  googleTagManager: {
    enabled: boolean;
  }
}
```